src/quick-lint-js/parse.h

```diff
@@ -2808,7 +2808,16 @@ class parser {
 
     if (this->peek().type == token_type::kw_else) {
       this->skip();
+      const char8 *end_of_else = this->lexer_.end_of_previous_token();
+      bool has_left_paren = this->peek().type == token_type::left_paren;
       parse_and_visit_body();
```

Comment:
Idea to fix error_missing_semicolon_after_statement:
```suggestion
      if (has_left_paren) {
        // The body of the 'else' is an expression.
        // copy-pasta from https://github.com/quick-lint/quick-lint-js/blob/7d32d9806fc70cc29bcaf5eaea921302860f9418/src/quick-lint-js/parse.h#L394-L395
        this->parse_and_visit_expression(v);
        // Don't do this: parse_expression_end();  // calls this->consume_semicolon();, which is what's reporting the extraneous error
      } else {
        parse_and_visit_body();
      }
```